Src/libCZI/CziAttachmentsDirectory.cpp

```diff
@@ -200,12 +176,12 @@ bool CReaderWriterCziAttachmentsDirectory::TryRemoveAttachment(int key, Attachme
 
 bool CReaderWriterCziAttachmentsDirectory::TryAddAttachment(const AttachmentEntry& entry, int* key)
 {
-    for (auto it = this->attchmnts.cbegin(); it != this->attchmnts.cend(); ++it)
+    if (std::find_if(
+        this->attchmnts.cbegin(),
+        this->attchmnts.cend(),
+        [&entry](const std::pair<int, AttachmentEntry>& x)->bool { return CCziAttachmentsDirectoryBase::CompareForEquality_Id(x.second, entry); }) != this->attchmnts.cend())
     {
-        if (CCziAttachmentsDirectoryBase::CompareForEquality_Id(it->second, entry))
-        {
-            return false;
-        }
+        return false;
```

Comment:
Do you want to add a test, which covers this case?

![image](https://github.com/ZEISS/libczi/assets/102169252/950e6f6f-b222-4b46-94fe-a67acb2012e3)

And maybe a test for the bugfix itself would be useful. (i.e. one that assures the order of the attachments)

Reply:
Added a test which exercises this line.

Regarding a test which ensures that the order in the Attachment-Directory is the same as attachments got added - I did not add such a thing for the following reasons:
- It would not be straight-forward to do so - there is no public/internal API which would give this information with guaranteed order.
- The order in question here is supposed to be strictly opaque - so I wouldn't see a value in providing an API for this purpose.
- I wouldn't want to advertise that "order is preserved" at all and make no guarantees here at all.